### PR TITLE
fix wrong kuberay image used by compatibility test

### PR DESCRIPTION
**Pull request by wilsonwang371**
## Why are these changes needed?

The current compatibility test does not use the correct build of kuberay. This patch fixed this issue

## Related issue number

#323

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by wilsonwang371** [comment]
cc @Jeffwan @DmitriGekhtman @scarlet25151

File: tests/compatibility-test.py [hunk 1]
```diff
@@ -46,7 +47,8 @@ def apply_kuberay_resources():
     shell_assert_success(
         'kubectl create -k manifests/cluster-scope-resources')
     shell_assert_success(
-        'kubectl create -k manifests/base')
+        'kustomize build manifests/base | sed "s/image: kuberay\\/\\(.*\\):nightly/image: kuberay\\/\\1:{}/g" | kubectl apply -f -'.format(
```
Comment (DmitriGekhtman) [inline tests/compatibility-test.py:50]: Would you mind adding a comment to this sed command?

**Review by DmitriGekhtman — APPROVED** [review]

File: tests/compatibility-test.py [hunk 2]
```diff
@@ -45,8 +46,10 @@ def apply_kuberay_resources():
     shell_assert_success('kind load docker-image kuberay/apiserver:{}'.format(kuberay_sha))
     shell_assert_success(
         'kubectl create -k manifests/cluster-scope-resources')
+    # use kustomize to build the yaml, then change the image to the one we want to testing.
     shell_assert_success(
-        'kubectl create -k manifests/base')
+        'kustomize build manifests/base | sed "s/image: kuberay\\/\\(.*\\):nightly/image: kuberay\\/\\1:{}/g" | kubectl apply -f -'.format(
```
Comment (Jeffwan) [inline tests/compatibility-test.py:51]: Can we use command like below? @wilsonwang371 

```
kubectl kustomize edit set image kuberay/operator=kuberay/operatorr:commit_hash
kubectl apply -k manifests/bases

```

Comment (wilsonwang371) [reply tests/compatibility-test.py]: done

**Review by Jeffwan — APPROVED** [review]

**Pull request merged**